### PR TITLE
Fix several bugs around handling of functions wrapped in proxies.

### PR DESCRIPTION
**Pull request by VSadov**
The proxies are callable (have [[Call]] method) and in some scenarios should defer to the underlying functions.

Fixes:https://github.com/Microsoft/ChakraCore/issues/5282
Fixes:https://github.com/Microsoft/ChakraCore/issues/4728
Fixes:https://github.com/Microsoft/ChakraCore/issues/4730

### CODE REVIEW

File: test/es6/ProxyInProxy.js [hunk 1]
```diff
@@ -172,7 +172,60 @@ function test4() {
...
+function test5() {
+    print("***function wrapped in 2+ proxies");
+
+    function a() {}
+
+    const p1 = new Proxy(a, {})
+    console.log(typeof p1) // "function"
+    
+    const p2 = new Proxy(p1, {})
+    console.log(typeof p2) // "object" but should be "function"
```
Comment (MSLaguana) [inline test/es6/ProxyInProxy.js]: Looks like the comments are stale, since this PR resolves the issue.

File: test/es6/ProxyInProxy.js [hunk 2]
```diff
@@ -172,7 +172,60 @@ function test4() {
...
+    
+    console.log(Object.prototype.toString.call(new Proxy(new Date, {})));
+    // "[object Object]"
+}
+
+function test7() {
+    print("*** proxied function and Function.prototype.toString.call");
+
+    console.log(Function.prototype.toString.call(new Proxy(function a() { }, {})));
+    // "[object Function]"
```
Comment (MSLaguana) [inline test/es6/ProxyInProxy.js]: These comments are also outdated, since the baseline suggests it actually returns the string of the function.

Comment (VSadov) [reply test/es6/ProxyInProxy.js]: Yes. I copied these from the original repro. They were referring to the behavior that needs to be changed.

**Comment by dilijev** [comment]
@VSadov did you mean to merge this to release/1.10?

**Pull request merged**